packages/server/graphql/mutations/helpers/notifications/SlackNotifier.ts

```diff
@@ -297,5 +297,121 @@ export const SlackNotifier: Notifier = {
 
   async integrationUpdated() {
     // Slack sends a system message on its own
+  },
+
+  async shareTopic(
+    dataLoader: DataLoaderWorker,
+    userId: string,
+    teamId: string,
+    meetingId: string,
+    reflectionGroupId: string,
+    stageIndex: number,
+    channelId: string
+  ) {
+    const r = await getRethink()
+    const [user, team, meeting, reflectionGroup, reflections, slackAuth] = await Promise.all([
+      dataLoader.get('users').load(userId),
+      dataLoader.get('teams').load(teamId),
+      dataLoader.get('newMeetings').load(meetingId),
+      dataLoader.get('retroReflectionGroups').load(reflectionGroupId),
+      r.table('RetroReflection').getAll(reflectionGroupId, {index: 'reflectionGroupId'}).run(),
+      r.table('SlackAuth').getAll(userId, {index: 'userId'}).filter({teamId})(0).run()
+    ])
+
+    if (!user) {
+      throw new Error('User not found')
+    }
+
+    if (!team) {
+      throw new Error('Team not found')
+    }
+
+    if (!slackAuth) {
+      throw new Error('Slack auth not found')
+    }
+
+    const {botAccessToken, slackUserId} = slackAuth
+    const manager = new SlackServerManager(botAccessToken!)
+    const channelInfo = await manager.getConversationInfo(channelId)
+
+    if (channelId !== slackUserId) {
+      if (!channelInfo.ok) {
+        throw new Error(channelInfo.error)
+      }
+      const {channel} = channelInfo
+      const {id: channelId, is_member: isMember, is_archived: isArchived} = channel
+      if (isArchived) {
+        throw new Error('Slack channel archived')
+      }
+      if (!isMember) {
+        const joinConvoRes = await manager.joinConversation(channelId)
+        if (!joinConvoRes.ok) {
+          throw new Error('Unable to join slack channel')
+        }
+      }
+    }
+
+    const topic = reflectionGroup.title
+    const discussionUrl = makeAppURL(appOrigin, `meet/${meetingId}/discuss/${stageIndex + 1}`)
+    const meetingUrl = makeAppURL(appOrigin, `meet/${meetingId}`)
+
+    const reflectionsText = reflections
```

Comment:
+1 should there be a limit to the reflection text? If a topic has many very long reflections, should we only show some of it?

Reply:
I'm not sure if we need to enforce the limit on the number of reflections. Feels like we are trying to read the user's mind, whether he wanted to keep 10 or 20 reflections to be shared. Slack has the limit of 3000 characters per block. I now check this and truncate the text in this case. Will it be sufficient for now?

Reply:
Yeah, I was thinking character limit rather than limiting the number of reflections. I think it makes sense to go with the same limit that Slack sets so 3,000 characters sounds good to me 👍